### ISSUE TITLE
Multiple metrics are not stored in Piwik 3.0 for plotting

### ISSUE THREAD

**Issue by aimeos** [matomo-org/matomo#11499]
After upgrading from the last Piwik 2 to Piwik 3.0.2, selecting multiple metrics in the "Visitor Overview" and the "Goal Overview" aren't saved. After a page reload, only the default graph is shown again.

**Comment by sgiehl** [comment]
@aimeos Are you viewing Piwik as anonymous user? It works for me if I'm logged in.

**Comment by aimeos** [comment]
@sgiehl No, I'm logged in too. Rearranging the dashboard is saved but not the metrics in the named panels. Can this be a problem rooted in the upgrade from Piwik 2?

**Comment by sgiehl** [comment]
@aimeos hm... that's strange. could you try to create a new dashboard with those widgets on it and try if it works there?

**Comment by aimeos** [comment]
@sgiehl Same problem

**Comment by mattab** [comment]
Can you please list exact steps to reproduce and what you get VS what you expected to see? Screenshot welcome too :+1:

**Comment by aimeos** [comment]
Login -> Dashboard:
![screenshot_1](https://cloud.githubusercontent.com/assets/8647429/24174218/d49eb118-0e8f-11e7-9b78-1128cfce3ac8.png)

Add "Pageviews" metric:
![screenshot_2](https://cloud.githubusercontent.com/assets/8647429/24174222/d8f6b7ec-0e8f-11e7-976d-f7721e3a99a4.png)

Click on "Dashboard" link in the left navigation:
![screenshot_1](https://cloud.githubusercontent.com/assets/8647429/24174218/d49eb118-0e8f-11e7-9b78-1128cfce3ac8.png)

As you can see, the "Pageviews" isn't plotted any more. Expected are the same metrics as before:
![screenshot_3](https://cloud.githubusercontent.com/assets/8647429/24174407/baa5c7b4-0e90-11e7-91f7-5a36cc019815.png)

Before, in Piwik 2.x the selected metrics where saved and even shown after logout/re-login.
This is the dashboard config in the piwik_user_dashboard table:
```
{"config":{"layout":"33-33-33"},"columns":[[{"uniqueId":"widgetVisitOverviewWithGraph","parameters":{"module":"CoreHome","action":"renderWidgetContainer","containerId":"VisitOverviewWithGraph","widget":1,"columns":["nb_visits","nb_pageviews"],"columns_to_display":["nb_visits","nb_pageviews"],"rows":[],"rows_to_display":[]},"isHidden":false},{"uniqueId":"widgetActionsgetPageUrls","parameters":{"module":"Actions","action":"getPageUrls","widget":1,"enable_filter_excludelowpop":"0","viewDataTable":"table","availableWidth":432,"availableHeight":671,"columns":"nb_visits"},"isHidden":false},{"uniqueId":"widgetLivewidget","parameters":{"module":"Live","action":"widget","widget":1},"isHidden":false}],[{"uniqueId":"widgetGoalsOverview","parameters":{"module":"CoreHome","action":"renderWidgetContainer","widget":1,"columns":["nb_conversions","conversion_rate"],"columns_to_display":["nb_conversions","conversion_rate"],"rows":[],"rows_to_display":[],"isFooterExpandedInDashboard":false,"containerId":"GoalsOverview"},"isHidden":false},{"uniqueId":"widgetReferrersgetWebsites","parameters":{"module":"Referrers","action":"getWebsites","widget":1,"isFooterExpandedInDashboard":true,"viewDataTable":"table","flat":"0"},"isHidden":false}],[{"uniqueId":"widgetUserCountryMapvisitorMap","parameters":{"module":"UserCountryMap","action":"visitorMap","widget":1},"isHidden":false},{"uniqueId":"widgetSEOgetRank","parameters":{"module":"SEO","action":"getRank","widget":1},"isHidden":false},{"uniqueId":"widgetVisitTimegetVisitInformationPerServerTime","parameters":{"module":"VisitTime","action":"getVisitInformationPerServerTime","widget":1,"columns":["nb_visits","nb_actions"],"columns_to_display":["nb_visits","nb_actions"],"rows":[],"rows_to_display":[]},"isHidden":false},{"uniqueId":"widgetDevicesDetectiongetBrowserEngines","parameters":{"module":"DevicesDetection","action":"getBrowserEngines","widget":1,"columns":["nb_visits"],"columns_to_display":["nb_visits"],"rows":[],"rows_to_display":[],"viewDataTable":"graphPie"},"isHidden":false},{"uniqueId":"widgetReferrersgetKeywords","parameters":{"module":"Referrers","action":"getKeywords","widget":1,"isFooterExpandedInDashboard":true,"viewDataTable":"table"},"isHidden":false},{"uniqueId":"widgetReferrersgetSearchEngines","parameters":{"module":"Referrers","action":"getSearchEngines","widget":1,"isFooterExpandedInDashboard":true,"viewDataTable":"table"},"isHidden":false}]]}
```

**Comment by peterbo** [comment]
This is also broken on the demo.piwik.org but not in all widget types. (e.g. visits over time is running well, visits overview (with graph) is broken.

Steps to reproduce:
- put e.g. the "visits overview (with graph)" widget to the dashboard
- plot another metric, e.g. bounce rate
- reload the dashboard
-> the bounce rate graph is not displayed

**Comment by tsteur** [comment]
Is this only about Dashboard? If not you might want to test https://github.com/piwik/piwik/pull/11983 and see if it changes anything

**Comment by peterbo** [comment]
Yes, the problem only appears on the dashboard (but only in some widgets). The PR at https://github.com/piwik/piwik/pull/11983 didn't fix it here.

**Comment by sgiehl** [comment]
I guess the problem only occurs in widgets that contain multiple other "widgets".
They are built using a widget container which loads the containing widgets. While the url for loading the widget container contains the saved settings as params, the urls for the widgets loaded afterwards don't contain the params anymore. So we need to find a way to give the params from container to the widgets it contains...

**Comment by olivierlambert** [comment]
Same issue for me (worked in Piwik 2, just upgraded to 3, it doesn't save my metrics I want to plot, just those by default (eg Visitors and not unique visitors). As soon as I refresh the widget, it puts me back to the default metrics.

**Comment by mattab** [comment]
@sgiehl maybe this is a regression we can address soon?

**Comment by urmet0** [comment]
Same issue here after upgrading from Piwik 2.x to 3.2.0.

**Comment by olivierlambert** [comment]
Seems fixed for me in `3.2.1`

**Comment by aimeos** [comment]
Upgraded to 3.2.1 but still the same issue
